File: api/WebServer.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// WebServer 对应 WebServer.py。
type WebServer struct {
	*session.ApiGroup
}

// NewWebServer 创建 WebServer API 分组。
func NewWebServer(s *session.Session) *WebServer {
	return &WebServer{ApiGroup: session.NewApiGroup(s)}
}

// Publickey 对应 webserver/publickey。
func (w *WebServer) Publickey() (map[string]interface{}, error) {
	return w.S.Get("webserver/publickey", nil, "api")
}

// Token 对应 webserver/token。
func (w *WebServer) Token() (map[string]interface{}, error) {
	return w.S.Get("webserver/token", nil, "api")
}

// WhiteListSwitch 对应 webserver/white-list-switch。
func (w *WebServer) WhiteListSwitch() (map[string]interface{}, error) {
	return w.S.Get("webserver/white-list-switch", nil, "api")
}

// SesTokInfo 对应 webserver/SesTokInfo。
func (w *WebServer) SesTokInfo() (map[string]interface{}, error) {
	return w.S.Get("webserver/SesTokInfo", nil, "api")
}

File: src/api/web_server.rs
//! WebServer API group (`api/WebServer.py`).
//!
//! Session/token endpoints that back the connection layer.

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// WebServer API group.
pub struct WebServer<'a> {
    conn: &'a Connection,
}

impl<'a> WebServer<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        WebServer { conn }
    }

    /// `webserver/publickey`. Public key for login encryption.
    pub fn publickey(&self) -> Result<Value> {
        get_value(self.conn, "api/webserver/publickey")
    }

    /// `webserver/token`. Session token.
    pub fn token(&self) -> Result<Value> {
        get_value(self.conn, "api/webserver/token")
    }

    /// `webserver/white_list_switch`. White-list switch.
    pub fn white_list_switch(&self) -> Result<Value> {
        get_value(self.conn, "api/webserver/white_list_switch")
    }

    /// `webserver/SesTokInfo`. Session-token info (valid for e.g. B310s-22).
    pub fn ses_tok_info(&self) -> Result<Value> {
        get_value(self.conn, "api/webserver/SesTokInfo")
    }
}
